File: VigenereCipher.java
import java.util.Scanner;

class VigenereCipher {
	public static String letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
	public static String key = "BELIEVE";
	public static int keyLength = key.length();

	public static String encrypt(String text) {
		text = text.toUpperCase();
		String cipherText = "";
		char ch, k;
		int index;
		for (int i = 0; i < text.length(); i++) {
			ch = text.charAt(i);
			k = key.charAt(i % keyLength);
			if (ch >= 'A' && ch <= 'Z') {
				index = (letters.indexOf(ch) + letters.indexOf(k)) % 26;
				cipherText += letters.charAt(index);
			} else {
				cipherText += ch;
			}
		}

		return cipherText;
	}

	public static String decrypt(String text) {
		text = text.toUpperCase();
		String plainText = "";
		char ch, k;
		int index;
		for (int i = 0; i < text.length(); i++) {
			ch = text.charAt(i);
			k = key.charAt(i % keyLength);
			if (ch >= 'A' && ch <= 'Z') {
				index = (letters.indexOf(ch) - letters.indexOf(k) + 26) % 26;
				plainText += letters.charAt(index);
			} else {
				plainText += ch;
			}
		}

		return plainText;
	}

	public static void main(String[] args) {
		// create a scanner object to read the user input
		Scanner sc = new Scanner(System.in);

		// write infinite loop to keep the program running until the user exits
		while (true) {

			// first ask the user if they want to encrypt or decrypt
			System.out.print("Do you want to encrypt or decrypt? (E/D) :");
			String choice = sc.nextLine();

			if (choice.equals("E") || choice.equals("e")) {
				// ask the user to enter the text to be encrypted
				System.out.print("Enter the text to be encrypted: ");
				String text = sc.nextLine();

				// encrypt the text and print the result
				String cipherText = encrypt(text);

				// encrypt the text and print the result
				System.out.println("Encrypted text: " + cipherText);

				// verification
				String decryptedText = decrypt(cipherText);
				System.out.println("Decrypted text: " + decryptedText);
			} else if (choice.equals("D") || choice.equals("d")) {
				// ask the user to enter the text to be decrypted
				System.out.print("Enter the text to be decrypted: ");
				String text = sc.nextLine();

				// decrypt the text and print the result
				String plainText = decrypt(text);

				// encrypt the text and print the result
				System.out.println("Decrypted text: " + plainText);

				// verification
				String encryptedText = encrypt(plainText);
				System.out.println("Encrypted text: " + encryptedText);
			}

			// ask the user if they want to continue
			System.out.print("Do you want to continue? (Y/N) :");
			choice = sc.nextLine();

			// if the user enters N or n, exit the program
			if (choice.equals("N") || choice.equals("n")) {
				System.out.println("Exiting...");
				break;
			}
		}
		sc.close();
	}
}

File: python-version/VigenereCipher.py
class VigenereCipher:
	letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
	key = "BELIEVE"
	keyLength = len(key)

	@staticmethod
	def encrypt(text):
		text = text.upper()
		cipherText = ""
		for i in range(len(text)):
			ch = text[i]
			k = VigenereCipher.key[i % VigenereCipher.keyLength]
			if ch.isalpha():
				index = (VigenereCipher.letters.index(ch) + VigenereCipher.letters.index(k)) % 26
				cipherText += VigenereCipher.letters[index]
			else:
				cipherText += ch
		return cipherText

	@staticmethod
	def decrypt(text):
		text = text.upper()
		plainText = ""
		for i in range(len(text)):
			ch = text[i]
			k = VigenereCipher.key[i % VigenereCipher.keyLength]
			if ch.isalpha():
				index = (VigenereCipher.letters.index(ch) - VigenereCipher.letters.index(k) + 26) % 26
				plainText += VigenereCipher.letters[index]
			else:
				plainText += ch
		return plainText

if __name__ == '__main__':
	# write infinite loop to keep the program running until the user exits
	while True:
		# first ask the user if they want to encrypt or decrypt
		choice = input("Do you want to encrypt or decrypt? (E/D) :")

		if choice.lower() == "e":
			# ask the user to enter the text to be encrypted
			text = input("Enter the text to be encrypted: ")

			# encrypt the text and print the result
			cipherText = VigenereCipher.encrypt(text)
			print("Encrypted text: " + cipherText)

			# verification
			decryptedText = VigenereCipher.decrypt(cipherText)
			print("Decrypted text: " + decryptedText)

		elif choice.lower() == "d":
			# ask the user to enter the text to be decrypted
			text = input("Enter the text to be decrypted: ")

			# decrypt the text and print the result
			plainText = VigenereCipher.decrypt(text)
			print("Decrypted text: " + plainText)

			# verification
			encryptedText = VigenereCipher.encrypt(plainText)
			print("Encrypted text: " + encryptedText)

		# ask the user if they want to continue
		choice = input("Do you want to continue? (Y/N) :")

		# if the user enters N or n, exit the program
		if choice.lower() == "n":
			print("Exiting...")
			break
